### ISSUE TITLE
Persist doesnt work with react-native-async-storage/async-storage

### ISSUE THREAD

**Issue by OsapBender** [ctrlplusb/easy-peasy#694]
Today I upgraded easy-peasy from v3.3.1 to v5.0.3 and my async store doesnt work anymore. 
I have persistor
```
import AsyncStorage from '@react-native-async-storage/async-storage';

import {persist} from 'easy-peasy';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
export default (reducer) =>
  persist(reducer, {
    storage: {
      async getItem(key) {
        return JSON.parse(await AsyncStorage.getItem(key));
      },
      setItem(key, data) {
        const jsonValue = JSON.stringify(data);
        console.log('key2: ', key, jsonValue);
        AsyncStorage.setItem(key, jsonValue);
      },
      removeItem(key) {
        console.log('key3: ', key);
        AsyncStorage.removeItem(key);
      },
    },
  });

```

on v3.3.1 my console log was like
<img width="515" alt="Снимок экрана 2021-08-29 в 17 04 51" src="https://user-images.githubusercontent.com/43818081/131251388-3b8f0b84-50e8-4685-ad79-a0965df6b95b.png">

but after upgrading it looks like
<img width="518" alt="Снимок экрана 2021-08-29 в 17 06 44" src="https://user-images.githubusercontent.com/43818081/131251377-0a69ba97-871e-43b9-aee9-2bce342bee5d.png">

my store model:
```
import {Action, action} from 'easy-peasy';
import persistor from '../persistor';
import {keys, storeAsyncData} from '../asyncStorage';

export interface IAuthModel {
  isAuth: boolean;
  setAuth: Action<IAuthModel, boolean>;
}

const authModel: IAuthModel = {
  isAuth: false,
  setAuth: action((state, payload: boolean): void => {
    state.isAuth = payload;
  }),
};

const thunks = {};

export default persistor({
  ...authModel,
  ...thunks,
});

```

```
import AppModel, {IAppState} from './app';
import Auth from './auth';
import {IAuthModel} from './auth';

export interface IStoreModel {
  app: IAppState;
  auth: IAuthModel;
}

const storeModel: IStoreModel = {
  app: AppModel,
  auth: Auth,
};

export default storeModel;

```

if i downgrade easy-peasy from 5.0.3 to 3.3.1 it works fine.

What's wrong?

PS
I read that changes from 3.3.1 to 5.0.3 but cannot find a solution

**Comment by rpadilla6** [comment]
Confirmed I had this issue when trying to upgrade to v5, I'm stuck at v3 for this reason. If I have the luxury of bonus time over the holidays maybe I can take a look and see what's going on

**Comment by rbruels** [comment]
This may be the issue for anyone reading this -- certainly was for me after the v3 -> v5 upgrade. There is a workaround in this thread.
      https://github.com/ctrlplusb/easy-peasy/issues/599
The root cause (maddening):
      https://github.com/facebook/react-native/issues/28602

**Comment by sefaokumus** [comment]
Is there any progress on this issue ?

**Comment by damassi** [comment]
FWIW, we've been using this same setup in easy-peasy@5.2 without problem. And just upgraded to v6 with no problem.

**Comment by no-stack-dub-sack** [comment]
Thanks @damassi! Is it possible to post a simple example of the setup that's working for you in case anyone else is blocked by this? As suggested by @rbruels the root cause may have been an RN regression issue, so I'm wondering if maybe this is fixed in newer versions (though it doesn't appear to have been addressed on the linked issue)?

**Comment by damassi** [comment]
Coming back to this, I realized [we were already using](https://github.com/artsy/energy/blob/main/src/system/store/GlobalStore.tsx#L13) the `requestIdleCallback = null` hack, and that is why things were working! so working around the regression.

**Comment by no-stack-dub-sack** [comment]
Thanks for checking @damassi!

**Comment by jmyrland** [comment]
> Coming back to this, I realized [we were already using](https://github.com/artsy/energy/blob/main/src/system/store/GlobalStore.tsx#L13) the `requestIdleCallback = null` hack, and that is why things were working! so working around the regression.

This is now fixed in `v6.0.1`, which means that the `requestIdleCallback = null` workaround is no longer required for `easy-peasy`.
